doajtest/unit/test_tasks_ingestCrossrefArticles.py

```diff
@@ -0,0 +1,1748 @@
+from doajtest.helpers import DoajTestCase
+
+from portality.tasks import ingestarticles
+from doajtest.fixtures.article_crossref import CrossrefArticleFixtureFactory
+from doajtest.fixtures.accounts import AccountFixtureFactory
+
+import urlparse
+import time
+from portality.crosswalks import article_crossref_xml
+from portality.bll.services import article as articleSvc
+
+from portality import models
+from portality.core import app
+
+from portality.background import BackgroundException
+
+import ftplib, os, requests
+from urlparse import urlparse
+from lxml import etree
+
+DEFAULT_MAX_REMOTE_SIZE=262144000
+CHUNK_SIZE=1048576
+
+GET = requests.get
+
+
+class MockFileUpload(object):
```

Comment:
I know you've just copied this from an existing test, so this is entirely reasonable.  I have recently been trying to refactor the mocks into something like a mock factory, and it would be good if we can go that way with this new test class.  Take a look in something like https://github.com/DOAJ/doaj/blob/develop/doajtest/mocks/bll_article.py for examples